go/vt/vttablet/tabletmanager/vreplication/vcopier_test.go

```diff
@@ -1361,3 +1362,81 @@ func TestPlayerCopyTablesWithGeneratedColumn(t *testing.T) {
 		{"bbb2", "bbb", "20"},
 	})
 }
+
+func TestCopyTablesWithInvalidDates(t *testing.T) {
+	defer deleteTablet(addTablet(100))
+
+	execStatements(t, []string{
+		"create table src1(id int, dt date, primary key(id))",
+		fmt.Sprintf("create table %s.dst1(id int, dt date, primary key(id))", vrepldb),
+		"insert into src1 values(1, '2020-01-12'), (2, '0000-00-00');",
+	})
+
+	// default mysql flavor allows invalid dates: so disallow explicitly for this test
+	if err := env.Mysqld.ExecuteSuperQuery(context.Background(), "SET @@global.sql_mode=CONCAT(@@global.sql_mode, ',NO_ZERO_DATE,NO_ZERO_IN_DATE')"); err != nil {
+		fmt.Fprintf(os.Stderr, "%v", err)
+	}
+	defer func() {
+		if err := env.Mysqld.ExecuteSuperQuery(context.Background(), "SET @@global.sql_mode=REPLACE(@@global.sql_mode, ',NO_ZERO_DATE,NO_ZERO_IN_DATE','')"); err != nil {
```

Comment:
MySQL will re-arrange the `sql_mode` tokens as it sees fit. Earlier we appended `',NO_ZERO_DATE,NO_ZERO_IN_DATE'` to the end of `sql_mode`, but there's no guarantee that reading `sql_mode` the two will still be last, or in that order. 

The correct and safe way to remove the two directives is:

```sql
set @@session.sql_mode=REPLACE(REPLACE(@@session.sql_mode, 'NO_ZERO_DATE', ''), 'NO_ZERO_IN_DATE', '');
```

Note that there is no need to remove the comma separator.